### PR TITLE
This checkin removes un-necessary code reported by findbugs, these ch…

### PR DESCRIPTION
**Pull request by anurag-gujral**
…anges have been discussed with Sanjeev

### CODE REVIEW

**Comment by FrankGasparovic** [comment]
This branch covers two stories: 
US40120
US60651

File: acs-integration-tests/src/test/java/com/ge/predix/integration/test/SubjectResourceFixture.java [hunk 1]
```diff
@@ -18,7 +18,11 @@
 import com.ge.predix.acs.rest.BaseResource;
 import com.ge.predix.acs.rest.BaseSubject;
 
-public class SubjectResourceFixture {
+public final class SubjectResourceFixture {
+    
+    private SubjectResourceFixture() {
+        //not called
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/integration/test/SubjectResourceFixture.java:24]: best to throw an exception if construction is attempted by a method of this class:

throw new IllegalAccessError("Class is non-instantiable");

File: acs-integration-tests/src/test/java/com/ge/predix/test/TestConfig.java [hunk 2]
```diff
@@ -5,7 +5,11 @@
 
 import com.ge.predix.acs.AccessControlService;
 
-public class TestConfig {
+public final class TestConfig {
+    
+    private TestConfig() {
+        //not called
```
Comment (anubhavi25) [inline acs-integration-tests/src/test/java/com/ge/predix/test/TestConfig.java]: see above

File: checkstyle-config/gog-sun-checks-eclipse.xml [hunk 3]
```diff
@@ -87,10 +87,6 @@
     </module>
     <module name="EmptyStatement"/>
     <module name="EqualsHashCode"/>
-    <module name="HiddenField">
```
Comment (anubhavi25) [inline checkstyle-config/gog-sun-checks-eclipse.xml:90]: need to reach a consensus on whether or not to remove this CheckStyle rule.

**Review by anubhavi25 — COMMENTED** [review]
Looks good for the most part 👍

**Comment by FrankGasparovic** [comment]
Build Success

[Build Status](https://predix1.jenkins.build.ge.com/job/Predix-Security/job/acs-build-n-test-pullrequest/718/)

File: acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java [hunk 4]
```diff
@@ -46,6 +46,7 @@
  *
  * @author 212338046
  */
+//CHECKSTYLE:OFF
```
Comment (anurag-gujral) [inline acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java:49]: why checkstyle is off here is it because it is a test?

File: commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java [hunk 5]
```diff
@@ -84,6 +85,7 @@ public void getSubdomainEmptyDomain() {
         Assert.assertNull(HttpServletRequestUtil.getSubdomain(hostname, domain));
     }
 
+    @SuppressFBWarnings(value = "NP_NULL_PARAM_DEREF_NONVIRTUAL")
```
Comment (anurag-gujral) [inline commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java]: We will have to remove @SuppressFBWarnings even though it means false positive from findbugs report since our black duck scan team is disallowing findBugs annotations usage in our code. I need to talk to Jeff from black duck scan team why they are disallowing findBugs annotation until then we will have to assume that findBugs annotations are dis-allowed.

File: model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java [hunk 6]
```diff
@@ -43,6 +41,10 @@
 
     private LinkedHashSet<String> policySetsEvaluationOrder = EMPTY_POLICY_EVALUATION_ORDER;
 
+    public static LinkedHashSet<String> getEmptyPolicyEvaluationOrder() {
+        return EMPTY_POLICY_EVALUATION_ORDER;
```
Comment (anurag-gujral) [inline model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java]: we are passing access to private variable through this function, do we really want to do this?

File: pom.xml [hunk 7]
```diff
@@ -146,6 +148,11 @@
             <!-- <version>${jackson.version}</version> -->
         </dependency>
         <dependency>
+            <groupId>com.google.code.findbugs</groupId>
```
Comment (anurag-gujral) [inline pom.xml]: black ducks scan team at GE does not want us to use findBugs as dependency please remove it.

File: service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java [hunk 8]
```diff
@@ -311,25 +310,25 @@ private String timestampValue() {
         }
     }
 
-    abstract void delete(final String key);
+    abstract void delete(String key);
```
Comment (anurag-gujral) [inline service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java:313]: Can you please tell why are we removing the final keyword?

File: model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java (hunk 6 above)
Comment (anubhavi25) [reply model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java]: we should remove this method since i'm removing LinkedHashSet in favor of the Set interface in another feature branch

File: service/src/main/java/com/ge/predix/acs/privilege/management/dao/SubjectRepository.java [hunk 9]
```diff
@@ -16,15 +16,14 @@
...
+import com.ge.predix.acs.zone.management.dao.ZoneEntity;
 import org.springframework.data.jpa.repository.JpaRepository;
 
-import com.ge.predix.acs.zone.management.dao.ZoneEntity;
+import java.util.List;
 
 public interface SubjectRepository extends JpaRepository<SubjectEntity, Long> {
 
-    List<SubjectEntity> findByZone(final ZoneEntity zone);
+    List<SubjectEntity> findByZone(ZoneEntity zone);
```
Comment (anurag-gujral) [inline service/src/main/java/com/ge/predix/acs/privilege/management/dao/SubjectRepository.java:26]: at many places we are removing final keyword, can you please explain why?

File: service/src/test/java/com/ge/predix/acs/policy/evaluation/cache/PolicyEvaluationRequestCacheKeyTest.java [hunk 10]
```diff
@@ -71,27 +71,42 @@ public void testBuildByRequestAndPolicySetEvaluationOrder() {
...
-                .request(request).policySetIds(request.getPolicySetsEvaluationOrder()).build();
+        PolicyEvaluationRequestCacheKey policyEvaluationRequestCacheKey =
+            new PolicyEvaluationRequestCacheKey.Builder().zoneId(ZONE_NAME)
+                                                         .request(request)
+                                                         .policySetIds(request.getPolicySetsEvaluationOrder())
+                                                         .build();
     }
 
+    @SuppressWarnings("unused")
+    @SuppressFBWarnings(value = "DLS_DEAD_LOCAL_STORE")
```
Comment (anurag-gujral) [inline service/src/test/java/com/ge/predix/acs/policy/evaluation/cache/PolicyEvaluationRequestCacheKeyTest.java]: Remove finbugs annotations

File: acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java (hunk 4 above)
Comment (anubhavi25) [reply acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java:49]: we should probably add notes everywhere "CHECKSTYLE:OFF" and "CHECKSTYLE:ON" are used mentioning why we're doing so

File: commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java (hunk 5 above)
Comment (anubhavi25) [inline commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java]: if we do remove this annotation, we should also remove the "com.google.code.findbugs.annotations" dependency from POMs

File: service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java (hunk 8 above)
Comment (anubhavi25) [reply service/src/main/java/com/ge/predix/acs/policy/evaluation/cache/AbstractPolicyEvaluationCache.java:313]: @anurag-gujral "final" is redundant when used for parameters in abstract method signatures (since the concrete method should add "final")

File: service/src/main/java/com/ge/predix/acs/privilege/management/dao/SubjectRepository.java (hunk 9 above)
Comment (anubhavi25) [reply service/src/main/java/com/ge/predix/acs/privilege/management/dao/SubjectRepository.java:26]: @anurag-gujral "final" is redundant when used for parameters in interface method signatures (since the concrete method should add "final")

File: model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java [hunk 11]
```diff
@@ -15,21 +15,19 @@
...
+import org.apache.commons.lang.builder.HashCodeBuilder;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
 
 @SuppressWarnings({ "javadoc", "nls" })
 @ApiModel(description = "Policy evaluation request for V1.")
 public class PolicyEvaluationRequestV1 {
-    public static final LinkedHashSet<String> EMPTY_POLICY_EVALUATION_ORDER = new LinkedHashSet<String>();
+    private static final LinkedHashSet<String> EMPTY_POLICY_EVALUATION_ORDER = new LinkedHashSet<String>();
```
Comment (anubhavi25) [inline model/src/main/java/com/ge/predix/acs/rest/PolicyEvaluationRequestV1.java]: we should revert this change since i'm removing LinkedHashSet in favor of the Set interface in another feature branch

File: acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java (hunk 4 above)
Comment (FrankGasparovic) [reply acs-integration-tests/src/test/java/com/ge/predix/acceptance/test/policy/admin/PolicyCreationStepsDefinitions.java:49]: Checkstyle is off because these cucumber tests don't follow the naming conventions in checkstyle.

File: commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java (hunk 5 above)
Comment (FrankGasparovic) [reply commons/src/test/java/com/ge/predix/acs/utils/HttpServletRequestUtilTest.java]: OK

File: pom.xml (hunk 7 above)
Comment (FrankGasparovic) [reply pom.xml]: OK

**Comment by FrankGasparovic** [comment]
Build success

[Build Status](https://predix1.jenkins.build.ge.com/job/Predix-Security/job/acs-build-n-test-pullrequest/724/)
